### ISSUE TITLE
"empty" values in dropdown fields

### ISSUE THREAD

**Issue by timkelty** [markhuot/craftql#129]
I think you're "faking" this because enums can't be empty…https://github.com/markhuot/craftql/blob/950a6410e25fba8ffb3a14e21343050a6a29c8c0/src/Helpers/StringHelper.php#L28-L37

Can't they be `null`? Seems like that might be a better choice.

**Comment by markhuot** [comment]
Unfortunately an enum value's name can not be null… GraphQL, essentially, tracks enums like this,

```php
$enum = [
  'name' => 'value',
  'fall' => 'fall',
  'SPRING' => 'spring',
  'All Year' => ''
]
```

The `name` from above is what you type in to a GraphQL query, such as,

```graphql
{
  entry(enumField: name) {
    #...
  }
}
```

From the GraphQL spec the `name` _must_ be a string. So, in the current configuration we end up with,

```graphql
{
  addSport(season: empty) {
    #...
  }
}
```

In the above think of `empty` not as a string, `"empty"`, but as a keyword for `empty`, if that makes sense.

--

Note: there was actually a bug here with how `empty` values were handled on upsets that is fixed on `dev-master` now.

**Comment by markhuot** [comment]
I'm going to leave this open until a stable tag is created.

**Comment by markhuot** [comment]
Closing this issue. The fix is included in version 1.2.1.
